message PBMsgUnassignHero
{
    required int32 cityIndex = 1;
    required int32 sid = 2;
    required int64 hid = 3;

    message PBMsgSkillActived {
        required int32 id = 1;
        required bool actived = 2;
        optional int32 lv = 3;
        optional int32 hero = 4;
        optional int32 gear = 5;
        optional int32 effectId = 6;
        repeated string effect = 7;
    }

    enum PBMsgHeroStatus {
        UNLOCKED = 1;
        IN_CITY_DO_NOTHING = 2;
        IN_CITY_IN_MARCH = 3;
        IN_CITY_SLEEP = 4;
        IN_NO_CITY = 5;
    }

    message PBMsgHero {
        required int64 userHeroId = 1;
        optional PBMsgHeroStatus status = 2;
        optional int32 heroId = 3;
        optional int32 level = 4;
        optional int32 renown = 5;
        optional int32 nextRenown = 6;
        optional int32 attack = 7;
        optional int32 hp = 8;
        optional int32 load = 9;
        repeated PBMsgSkillActived skill = 10;
        repeated PBMsgSkillActived fate = 11;
        optional int32 might = 12;
        optional int32 maxExplore = 13;
        optional int32 explored = 14;
        optional int32 sleepTime = 15;
        optional int32 cityIndex = 16;
        optional int32 slot = 17;
    }

    repeated PBMsgHero hero = 4;
}
